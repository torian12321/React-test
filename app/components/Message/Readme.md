#### Basic
```
<Message>Content</Message>
```

#### Types
```
<div>
	<Message type='info'>Info content</Message>
	<Message type='success'>Success content</Message>
	<Message type='danger'>Danger content</Message>
	<Message type='warning'>Warning content</Message>
</div>
```

#### Important
Show a shadow
```
<div>
	<Message>Some text</Message>
	<Message>Some text</Message>
	<Message important>Some text</Message>
	<Message>Some text</Message>
</div>
```