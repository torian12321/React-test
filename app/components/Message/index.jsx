var
React     = require('react'),
PropTypes = require('prop-types'),
classNames= require('classnames'),
styles    = require('./_style.less');

const Message = props => (
    <div className={classNames('message', props.type)}>
        {props.children}
    </div>
);


Message.propTypes = {
    children: PropTypes.any.isRequired,
    type    : PropTypes.oneOf(['info', 'success', 'danger', 'warning'])
};

module.exports = Message;
