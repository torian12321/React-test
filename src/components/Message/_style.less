@import '../../styles/conf/index.less';

.message{
	display          : block;
	margin           : 20px 0;
	padding          : 20px 10px;
	background       : @white;
	border           : solid 1px darken(@themeColor_1, 10%);
	border-left-width: 10px;
	.rounded();
	&.important{
		.shadow(@shadow);
	}

	// Styles
	&.info   { .message_palette(@colorState_info); }
	&.success{ .message_palette(@colorState_success); }
	&.warning{ .message_palette(@colorState_warning); }
	&.danger { .message_palette(@colorState_danger); }
}

.message_palette(@color: @themeColor_1){
	background  : lighten(@color, 30%);
	border-color: @color;
}