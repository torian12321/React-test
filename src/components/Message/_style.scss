@import '../../styles/conf/index.scss';

// Vars
$message-color-border: $themeColor_1 !default;
$message-color-bg    : $white !default;

// Block
.message{
  display    : block;
  margin     : 20px 0;
  padding    : 20px 10px;
  border     : $message-color-border $BStyle $BWidth;
  border-left: $message-color-border solid 10px;
  background : $message-color-bg;
  
  @include rounded();

  // Modifiers
  &--important{
    @include shadow();
  }

  @each $stateName, $stateColor in $states {
    &--#{$stateName} {
      background  : nth($stateColor, 2);
      border-color: nth($stateColor, 1);
    }
  }
}
