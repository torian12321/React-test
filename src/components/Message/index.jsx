var
React     = require('react'),
PropTypes = require('prop-types'),
classNames= require('classnames'),
styles    = require('./_style.less');

const Message = props => (
    <div className={classNames('message', props.type, {"important" : props.important})}>
        {props.children}
    </div>
);


Message.propTypes = {
    children : PropTypes.any.isRequired,
    type     : PropTypes.oneOf(['info', 'success', 'danger', 'warning']),
    important: PropTypes.bool
};
Message.defaultProps = {
    important: false
};

module.exports = Message;
