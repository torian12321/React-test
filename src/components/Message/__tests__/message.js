import React    from 'react';
import renderer from 'react-test-renderer';
import Message  from '../index';

describe('component', () => {
  describe('<Message />', () => {

    it('should render correctly', () => {
      expect(renderer.create(
        <Message>Lorem Ipsum</Message>
      )).toMatchSnapshot();
    });

    it('Should render danger type', () => {
      expect(renderer.create(
        <Message type='danger'>Lorem Ipsum</Message>
      )).toMatchSnapshot();
    });

    it('Should render important state', () => {
      expect(renderer.create(
        <Message important>Lorem Ipsum</Message>
      )).toMatchSnapshot();
    });

  });
});
