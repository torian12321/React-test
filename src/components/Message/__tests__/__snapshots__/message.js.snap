// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component <Message /> Should render danger type 1`] = `
<div
  className="message message--danger"
>
  Lorem Ipsum
</div>
`;

exports[`component <Message /> Should render important state 1`] = `
<div
  className="message message--important"
>
  Lorem Ipsum
</div>
`;

exports[`component <Message /> should render correctly 1`] = `
<div
  className="message"
>
  Lorem Ipsum
</div>
`;
